packages/postgres/src/sql-template.ts

```diff
@@ -12,14 +12,12 @@ export function sqlTemplate(
       "It looks like you tried to call `sql` as a function. Make sure to use it as a tagged template.\n\tExample: sql`SELECT * FROM users`, not sql('SELECT * FROM users')",
     );
   }
-  let result = '';
 
-  for (let i = 0; i < strings.length; i++) {
-    result += strings[i];
+  // ts tries to annotate `strings` with `& any[]` because of the prior check which breaks the type
+  let result = (strings as TemplateStringsArray)[0] ?? '';
 
-    if (i < values.length) {
-      result += `$${i + 1}`;
-    }
+  for (let i = 1; i < strings.length; i++) {
+    result += `$${i}${(strings as TemplateStringsArray)[i] ?? ''}`;
```

Comment:
the guard here is `strings.length` already, that `?? ''` shouldn't be necessary but engines here should be smart enough, although I don't understand why TS needs any of these, when `TemplateStringsArray` should have already a `length` greater than `0` by default and also be a list of `string[]` but as this has been merged already, I also don't care much, just wanted to leave my 2c in here.